base/markdown/GitHub/GitHub.jl

```diff
@@ -3,10 +3,12 @@ include("table.jl")
 @breaking true ->
 function fencedcode(stream::IO, block::MD, config::Config)
     startswith(stream, "```", padding = true) || return false
-    flavor = strip(readline(stream))
+    trailing = strip(readline(stream))
+    flavor = lstrip(trailing, '`')
+    n = 3 + length(trailing) - length(flavor)
```

Comment:
Inline use of fenced code blocks is also allowed with more than 1 \`, so I think if you have something like \`\`\`thing\`\`\` this needs to return false here (that is, `` "`" ^ n in flavor``) and catch in inline code.

eurgh so many edge cases http://johnmacfarlane.net/babelmark2/?text=%60%60%60%60asda%60%60%60%60%60

---

Also should support \~ but only for *block*... to catch `~~~` should I define a read method (to read up to n chars) or perhaps make it private e.g. `_readn`, if we'd rather not read the whole line. There doesn't seem to be one in base.

```
function read(io::IO, T::Type, n::Int)
     data = T[]
     for _ in 1:n
        eof(io) && break
        push!(data, read(io, T))
    end
    data
end
```